libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1003,11 +1003,24 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
             response.statusCode = StatusCodes.BAD_REQUEST;
             response.setBody(`Request missing verb and/or path. Verb: ${ request.verb }. Path: ${ request.path }`);
             return response;
-        } 
+        }
 
         if (request.verb.toLocaleUpperCase() === GET && request.path.toLocaleLowerCase() === VERSION_PATH) {
             response.statusCode = StatusCodes.OK;
-            response.setBody({UserAgent: USER_AGENT});
+
+            if (!this.credentials.appId || !this.credentials.appPassword) {
+                response.setBody({UserAgent: USER_AGENT});
+                return response;
+            }
+
+            try {
+                response.setBody({UserAgent: USER_AGENT, Token: await this.credentials.getToken()});
```

Comment:
Let's rename "Token" to "BotToken" just to make it clear it's not a user token, since the other half of this line is UserAgent

Reply:
That's fine with me!